File: LinkedList/self_define_object.java
class Node {
    int data;
    Node next;

    // Constructor
    Node(int data1, Node next1) {
        this.data = data1;
        this.next = next1;
    }

    Node(int data1) {
        this.data = data1;
        this.next = null;

    }
}

public class self_define_object {
    public static void main(String[] args) {
        int[] arr = { 2, 3, 4, 5 };
        Node y = new Node(arr[0], null);
        System.out.println(y.data);

        System.out.println("The head of the LinkedList");
        Node head = Convert2Ll(arr);
        System.out.println(head.data);

        // traverse through the array
        Node temp = head;
        while (temp != null) {
            System.out.print(temp.data + " ");
            temp = temp.next;
        }

        // Length of linkedList
        int length = lengthofLL(head);
        System.out.println();
        System.out.println("The length of the LinkedList is" + length);

        // Delete The Head
        head = DeleteHead(head);
        System.out.println();
        // System.out.println(head);
        Node temp2 = head;
        while (temp2 != null) {
            System.out.print(temp2.data + " ");
            temp2 = temp2.next;

        }
        
        //Delete Last Element\
        System.out.println();
        head = DeleteLast(head);
        Node temp3 = head;
        while (temp3 != null) {
            System.out.print(temp3.data + " ");
            temp3 = temp3.next;

        }

    }

    public static int lengthofLL(Node head) {
        int length = 0;
        Node temp = head;
        while (temp != null) {
            length++;
            temp = temp.next;
        }
        return length;
    }

    public static Node Convert2Ll(int[] arr) {
        Node head = new Node(arr[0]);
        Node mover = head;
        for (int i = 1; i < arr.length; i++) {
            Node temp = new Node(arr[i]);
            mover.next = temp;
            mover = temp;
        }
        return head;
    }

    public static Node DeleteHead(Node head) {
        if (head == null) {
            return head;

        } else {
            // Update head of the LL
            head = head.next;

        }
        return head;
    }

    public static Node DeleteLast(Node head){
        if(head == null|| head.next ==null) return null;
        Node temp = head ;
        while(temp.next.next!=null){
            temp = temp.next;
        }
        temp.next=null;


        return head;

}
}

File: LinkedList/self_define_object.cpp
#include <bits/stdc++.h>
using namespace std;

class Node
{
public:
    int data;
    Node *next;
    Node(int data1, Node *next1)
    {
        data = data1;
        next = next1;
    }

    Node(int data1)
    {
        data = data1;
        next = nullptr;
    }
};

Node *Convert2LL(vector<int> &arr)
{
    Node *head = new Node(arr[0]);
    Node *mover = head;
    for (int i = 1; i < arr.size(); i++)
    {
        Node *temp = new Node(arr[i]);
        mover->next = temp;
        mover = temp;
    }
    return head;
}
// Find The Length
int lengthOfLL(Node *head)
{
    int count = 0;
    Node *temp = head;
    while (temp != nullptr)
    {
        count++;
        temp = temp->next;
    }
    return count;
}
// Delete The Head of the LinkedList
Node *DeleteHead(Node *head)
{
    Node *temp = head;
    // Here Head point next temp and That is our new Head And after that we need to remove The Previous head Node
    head = head->next;
    free(temp);
    // delete(temp);
    return head;
}

// DeleteLast
Node *Deltelast(Node *head)
{
    if (head == NULL || head->next == NULL)
    {
        return NULL;
    }
    else{
        Node*temp = head;
        while(temp->next->next!=NULL){
            temp = temp->next;
        }
         delete temp->next;
        temp->next = NULL;

    }
    return head;
}

int main()
{
    vector<int> arr = {2, 3, 4, 5};
    Node *y = new Node(arr[0], nullptr);
    cout << y->data;
    cout << "\n";
    Node *head = Convert2LL(arr);
    cout << head->data;

    cout << "\n";
    Node *temp = head;
    while (temp)
    {
        cout << temp->data << " ";
        temp = temp->next;
    }
    // Find The Length of the LinkedList
    int length = lengthOfLL(head);
    cout << "\n";
    cout << "The Length of the LinkedList is" << " " << length;

    // delete hEad
    head = DeleteHead(head);
    cout << "\n";
    printf("The new head is ");
    cout << head->data;
    //DeleteLast
    cout<<"\n";
    cout<<"Deleting The Last Node"<<" ";
    head = Deltelast(head);
    Node*temp1 = head;
    while(temp1){
            cout << temp1->data << " ";
            temp1 = temp1->next;
    }
}